packages/fetchai/connections/http_server/connection.py

```diff
@@ -525,20 +525,25 @@ async def connect(self) -> None:
 
         :return: None
         """
-        if not self.connection_status.is_connected:
-            self.channel.logger = self.logger
-            await self.channel.connect(loop=self.loop)
-            self.connection_status.is_connected = not self.channel.is_stopped
+        if self.is_connected:
+            return
+        self._state.set(ConnectionStates.connecting)
+        self.channel.logger = self.logger
+        await self.channel.connect(loop=self.loop)
+        if not self.channel.is_stopped:
+            self._state.set(ConnectionStates.connected)
```

Comment:
if `self.channel.is_stopped`, the connection will remain in `connecting` state. Is it correct? Maybe we should set it to disconnected, or raise `ConnectionError`.

Reply:
It's also a matter of making it clearer at interface level what's the expected behaviour in such cases. Worth an issue imo.

Reply:
nice point, thanks. it was previous implementation can looks this was incorrect solution

updated. now it will be disconnected